registry/endpoint.go

```diff
@@ -171,12 +168,12 @@ func (e *Endpoint) pingV1(factory *requestdecorator.RequestFactory) (RegistryInf
 		return RegistryInfo{Standalone: false}, nil
 	}
 
-	req, err := factory.NewRequest("GET", e.Path("_ping"), nil)
+	req, err := http.NewRequest("GET", e.Path("_ping"), nil)
 	if err != nil {
 		return RegistryInfo{Standalone: false}, err
 	}
 
-	resp, _, err := doRequest(req, nil, ConnectTimeout, e.IsSecure)
+	resp, err := HTTPClient(DefaultTransport(ConnectTimeout, e.IsSecure)).Do(req)
```

Comment:
That sounds okay: `http.Client` is cheap to instantiate. The expensive piece is the transport, since you may disassociate from the connection pool. Seems like there should be a common `http.Transport` instance to support pooling but I do understand there are issues with the certs are handled.

If this was the previous behavior, no reason to change it for this PR.